### PR TITLE
Select queues

### PR DESCRIPTION
**Pull request by Christina-84**
Work in progress on a feature to select multiple queues
Skärmavbild 2021-09-26 kl  18 38 14

### CODE REVIEW

File: static/queues.html [hunk 1]
```diff
@@ -140,9 +160,26 @@ <h3>Add queue</h3>
...
-          avalanchemq.table.renderCell(tr, 0, item.vhost)
-          avalanchemq.table.renderCell(tr, 1, queueLink)
-          avalanchemq.table.renderCell(tr, 2, features, 'center')
+
+          var checkbox = document.createElement('input')
+          checkbox.type='checkbox'
+          checkbox.setAttribute('data-vhost', encodeURIComponent(item.vhost))
+          checkbox.setAttribute('data-name', encodeURIComponent(item.name))
+
+          checkbox.addEventListener('change', function() {
```
Comment (carlhoerberg) [inline static/queues.html:169]: doesn't checkboxes have a native checked attribute? https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input/checkbox#attr-checked

File: static/queues.html [hunk 2]
```diff
@@ -151,17 +188,26 @@ <h3>Add queue</h3>
...
+        avalanchemq.table.renderCell(tr, 12, avalanchemq.helpers.formatNumber(item.message_stats.ack_details.rate), 'right')
       })
 
+      function selectedCounter() {
+        var selectedList = document.querySelectorAll('[data-select]');
+        if (selectedList.length >= 1) {
+              document.getElementById('selected-controls').style.visibility="visible"
+            } else {
+              document.getElementById('selected-controls').style.visibility="hidden"
+            }
```
Comment (carlhoerberg) [inline static/queues.html:208]: indentation seems off here

File: static/queues.html [hunk 3]
```diff
@@ -184,6 +230,67 @@ <h3>Add queue</h3>
       document.querySelector('#dataTags').onclick = e => {
         window.avalanchemq.helpers.argumentHelperJSON("arguments", e)
       }
+
+      document.querySelector("#deleteQueues").addEventListener('submit', function (evt) {
+        evt.preventDefault()
+
+        if (window.confirm('Are you sure? The selected queues are going to be deleted. Messages cannot be recovered after deletion.')) {
+          var selectedList = document.querySelectorAll('[data-select]')
```
Comment (carlhoerberg) [inline static/queues.html:238]: always use `const` or `let`, never `var`. var has weird/outdated scoping rules. in this case you want to use `const` as the variable will never change.

File: static/queues.html [hunk 4]
```diff
@@ -184,6 +230,67 @@ <h3>Add queue</h3>
...
+          var vhost; 
+          var name; 
+
+          for (let i = 0; i < selectedList.length; i++) {
+            vhost = selectedList[i].dataset.vhost
+            name = selectedList[i].dataset.name
```
Comment (carlhoerberg) [inline static/queues.html:244]:
```suggestion
            const vhost = selectedList[i].dataset.vhost
            const name = selectedList[i].dataset.name
```

File: static/queues.html [hunk 5]
```diff
@@ -184,6 +230,67 @@ <h3>Add queue</h3>
...
         window.avalanchemq.helpers.argumentHelperJSON("arguments", e)
       }
+
+      document.querySelector("#deleteQueues").addEventListener('submit', function (evt) {
+        evt.preventDefault()
+
+        if (window.confirm('Are you sure? The selected queues are going to be deleted. Messages cannot be recovered after deletion.')) {
+          var selectedList = document.querySelectorAll('[data-select]')
+          var vhost; 
+          var name;
```
Comment (carlhoerberg) [inline static/queues.html:240]:
```suggestion
```

File: static/queues.html [hunk 6]
```diff
@@ -184,6 +230,67 @@ <h3>Add queue</h3>
...
+          var selectedList = document.querySelectorAll('[data-select]')
+          var vhost; 
+          var name; 
+
+          for (let i = 0; i < selectedList.length; i++) {
+            vhost = selectedList[i].dataset.vhost
+            name = selectedList[i].dataset.name
+            const url = '/api/queues/' + vhost + '/' + name
+            avalanchemq.http.request('DELETE', url)
+            .then(() => { window.location = '/queues' })
```
Comment (carlhoerberg) [inline static/queues.html:247]: won't this cause a page reload, for each delete?

File: static/queues.html [hunk 7]
```diff
@@ -184,6 +230,67 @@ <h3>Add queue</h3>
...
+            name = selectedList[i].dataset.name
+            const url = '/api/queues/' + vhost + '/' + name + '/contents'
+            avalanchemq.http.request('DELETE', url)
+            .then(() => { avalanchemq.dom.toast('Queues purged!') })
+            .catch(avalanchemq.http.standardErrorHandler)
+          }
+        }
+      })
+
+      document.querySelector("#pauseQueues").addEventListener('submit', function (evt) {
```
Comment (carlhoerberg) [inline static/queues.html:273]: what about unpause/resume?

File: static/queues.html [hunk 8]
```diff
@@ -184,6 +230,67 @@ <h3>Add queue</h3>
...
+          for (let i = 0; i < selectedList.length; i++) {
+            vhost = selectedList[i].dataset.vhost
+            name = selectedList[i].dataset.name
+            const url = '/api/queues/' + vhost + '/' + name
+            avalanchemq.http.request('DELETE', url)
+            .then(() => { window.location = '/queues' })
+            .catch(avalanchemq.http.standardErrorHandler)
+          }
+        }
+
```
Comment (carlhoerberg) [inline static/queues.html:251]:
```suggestion
```

File: static/queues.html [hunk 9]
```diff
@@ -184,6 +230,67 @@ <h3>Add queue</h3>
...
+            const url = '/api/queues/' + vhost + '/' + name + '/pause'
+            avalanchemq.http.request('PUT', url)
+            .then(() => {
+              avalanchemq.dom.toast('Queues paused!')
+            })
+            .catch(avalanchemq.http.standardErrorHandler)
+          }
+        }
+      })
+
```
Comment (carlhoerberg) [inline static/queues.html:293]:
```suggestion
```

File: static/queues.html [hunk 10]
```diff
@@ -184,6 +230,67 @@ <h3>Add queue</h3>
...
+          var name; 
+
+          for (let i = 0; i < selectedList.length; i++) {
+            vhost = selectedList[i].dataset.vhost
+            name = selectedList[i].dataset.name
+            const url = '/api/queues/' + vhost + '/' + name + '/pause'
+            avalanchemq.http.request('PUT', url)
+            .then(() => {
+              avalanchemq.dom.toast('Queues paused!')
+            })
```
Comment (carlhoerberg) [inline static/queues.html:288]:
```suggestion
            .then(() => avalanchemq.dom.toast('Queues paused!'))
```

File: static/queues.html (hunk 6 above)
Comment (snichme) [reply static/queues.html:247]: We talked about this and it will be changed so that the pages isn't reloaded at all.

**Pull request closed without merging**